model/build/build.go

```diff
@@ -267,6 +270,28 @@ func (b *Build) SetCachedTaskFinished(taskID, status string, detail *apimodels.T
 	return nil
 }
 
+func (b *Build) UpdateCachedTasks(tasks []task.Task) error {
+	cache := b.Tasks
+	for i, cacheTask := range b.Tasks {
+		for _, realTask := range tasks {
+			if cacheTask.Id != realTask.Id {
```

Comment:
With the `!=` approach, after finding a match and setting values on 281-284, the inner loop continues. I think you want to break or invert the logic.

Reply:
fixed